Upgrade patternfly to version 5

aljesusg opened the pull request:
Fix https://github.com/kiali/kiali/issues/6446
Fix https://github.com/kiali/kiali/issues/6453

Actual behaviour

## Overview

![image](https://github.com/kiali/kiali/assets/3019213/54c1a7d9-2cdd-4e00-9116-2e96f78b3a5e)


## Graph

![image](https://github.com/kiali/kiali/assets/3019213/a25ad54c-1307-4c2c-91a0-30cc61aa1314)

## Lists

![image](https://github.com/kiali/kiali/assets/3019213/e1de504e-11e4-4db1-9278-4309c7decc96)

## Detail

![image](https://github.com/kiali/kiali/assets/3019213/fe5f7436-4870-427a-a8c0-e0683a8f74c3)


## Metrics

![image](https://github.com/kiali/kiali/assets/3019213/ab91a965-f776-45a5-8d8e-b1ee9f4d6fd4)


## Traces

![image](https://github.com/kiali/kiali/assets/3019213/45d7286f-ebc7-4cde-8268-f45dd1c42d74)


## Dark Mode toolbar
![Screenshot from 2023-08-21 11-36-34](https://github.com/kiali/kiali/assets/3019213/19c2ee6b-cf75-48b3-8d6f-2d5077bbb838)

### Light Mode
![Screenshot from 2023-08-21 11-38-17](https://github.com/kiali/kiali/assets/3019213/2bc47f58-04a0-4d88-be75-2ae5b7982dee)
### Dark Mode
![Screenshot from 2023-08-21 11-38-36](https://github.com/kiali/kiali/assets/3019213/ee8baa53-f8c9-4a4e-8b4b-c8c262dda0ab)

aljesusg commented on frontend/src/components/IstioWizards/WizardAnnotations.tsx:
```diff
@@ -11,7 +11,7 @@ import {
   TitleSizes
 } from '@patternfly/react-core';
-import { TableComposable, Tbody, Th, Thead, Tr } from '@patternfly/react-table';
+import { Table /* data-codemods */, Tbody, Th, Thead, Tr } from '@patternfly/react-table';
```
This was added by @patternfly/pf-codemods@latest let me check with pf team about it

aljesusg commented on frontend/src/components/VirtualList/VirtualList.tsx:
```diff
@@ -168,9 +169,7 @@ class VirtualListComponent<R extends RenderResource> extends React.Component<Vir
-                      <Title headingLevel="h5" size={TitleSizes.lg}>
-                        No {typeDisplay} found
-                      </Title>
+                      <EmptyStateHeader titleText={<>No{typeDisplay}found</>} headingLevel="h5" />
```
weird this was made by @patternfly/pf-codemods@latest  too good catch

aljesusg commented on frontend/src/pages/IstioConfigNew/AuthorizationPolicyForm/When/ConditionBuilder.tsx:
```diff
@@ -70,7 +71,7 @@ export class ConditionBuilder extends React.Component<Props, State> {
   }
 
-  onAddNewConditionKey = (key: string, _) => {
+  onAddNewConditionKey = (_, key: string) => {
```
That is due the event trigger from pf cmponent, the first param must be the "event" onclick , '_' in ts means ommit this var

aljesusg commented on frontend/src/pages/IstioConfigNew/AuthorizationPolicyForm/When/ConditionBuilder.tsx:
```diff
@@ -88,7 +89,7 @@ export class ConditionBuilder extends React.Component<Props, State> {
   };
 
-  onAddNewNotValues = (notValues: string, _) => {
+  onAddNewNotValues = (_, notValues: string) => {
```
The same

aljesusg commented on frontend/src/pages/IstioConfigNew/GatewayForm/ListenerList.tsx:
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { cellWidth, TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { cellWidth, Table /* data-codemods */, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
```
This was added by @patternfly/pf-codemods@latest let me check with pf team about it

aljesusg commented on frontend/src/pages/IstioConfigNew/GatewayForm/ServerList.tsx:
```diff
@@ -1,6 +1,6 @@
 import * as React from 'react';
 import { Server, ServerForm } from '../../../types/IstioObjects';
-import { cellWidth, Tbody, Td, Th, Thead, Tr, TableComposable } from '@patternfly/react-table';
+import { cellWidth, Tbody, Td, Th, Thead, Tr, Table /* data-codemods */ } from '@patternfly/react-table';
```
This was added by @patternfly/pf-codemods@latest let me check with pf team about it

aljesusg commented on frontend/src/pages/IstioConfigNew/PeerAuthenticationForm.tsx:
```diff
@@ -101,7 +112,7 @@ export class PeerAuthenticationForm extends React.Component<Props, PeerAuthentic
   }
 
-  onChangeWorkloadSelector = () => {
+  onChangeWorkloadSelector = (_event, _: boolean) => {
```
Same

aljesusg commented on frontend/src/pages/IstioConfigNew/PeerAuthenticationForm.tsx:
```diff
@@ -112,7 +123,7 @@ export class PeerAuthenticationForm extends React.Component<Props, PeerAuthentic
   };
 
-  onChangeAddPortMtls = () => {
+  onChangeAddPortMtls = (_event, _: boolean) => {
```
Same

aljesusg commented on frontend/src/pages/IstioConfigNew/RequestAuthenticationForm.tsx:
```diff
@@ -57,7 +57,7 @@ export class RequestAuthenticationForm extends React.Component<Props, RequestAut
   };
 
-  onChangeWorkloadSelector = () => {
+  onChangeWorkloadSelector = (_event, _: boolean) => {
```
Same

aljesusg commented on frontend/src/pages/IstioConfigNew/RequestAuthenticationForm.tsx:
```diff
@@ -68,7 +68,7 @@ export class RequestAuthenticationForm extends React.Component<Props, RequestAut
   };
 
-  onChangeJwtRules = () => {
+  onChangeJwtRules = (_event, _: boolean) => {
```
Same

hhovsepy reviewed (commented):
Did a smoke test of all pages, in general a padding and width issues on columns.
![Screenshot from 2023-08-09 12-06-42](https://github.com/kiali/kiali/assets/604313/c5337ed5-74c0-4997-aa84-8dd91311b2cd)
![Screenshot from 2023-08-09 11-55-26](https://github.com/kiali/kiali/assets/604313/103822bf-d2d9-40b9-9b1a-709880cd2c5c)
![Screenshot from 2023-08-09 11-42-50](https://github.com/kiali/kiali/assets/604313/317bc949-96ee-472b-81e7-a35103f6d7d0)
![Screenshot from 2023-08-09 11-41-47](https://github.com/kiali/kiali/assets/604313/d863ffdf-613a-48c6-b5f8-e118e22b3b25)
![Screenshot from 2023-08-09 11-40-45](https://github.com/kiali/kiali/assets/604313/a8828d79-be47-492d-9b3d-3b3aa1d43c72)

Tooltip before:
![Screenshot from 2023-08-09 11-53-27](https://github.com/kiali/kiali/assets/604313/8633b32d-9b98-4b84-b1aa-0ab1c4557520)
Tooltip after:
![Screenshot from 2023-08-09 11-53-12](https://github.com/kiali/kiali/assets/604313/566cedc2-5c77-4408-be22-e38b15d51e74)

When objects are not loaded yet, spaces are missing:
![Screenshot from 2023-08-09 11-47-59](https://github.com/kiali/kiali/assets/604313/a68c924a-4246-462b-9a28-79fd2be32696)


Istio config details page is not loading:
![Screenshot from 2023-08-09 12-06-00](https://github.com/kiali/kiali/assets/604313/d5c89c28-12e0-46c1-974e-9acf165c125b)

aljesusg commented:
> Did a smoke test of all pages, in general a padding and width issues on columns. ![Screenshot from 2023-08-09 12-06-42](https://user-images.githubusercontent.com/604313/259372963-c5337ed5-74c0-4997-aa84-8dd91311b2cd.png) ![Screenshot from 2023-08-09 11-55-26](https://user-images.githubusercontent.com/604313/259372973-103822bf-d2d9-40b9-9b1a-709880cd2c5c.png) ![Screenshot from 2023-08-09 11-42-50](https://user-images.githubusercontent.com/604313/259372985-317bc949-96ee-472b-81e7-a35103f6d7d0.png) ![Screenshot from 2023-08-09 11-41-47](https://user-images.githubusercontent.com/604313/259372990-d863ffdf-613a-48c6-b5f8-e118e22b3b25.png) ![Screenshot from 2023-08-09 11-40-45](https://user-images.githubusercontent.com/604313/259372993-a8828d79-be47-492d-9b3d-3b3aa1d43c72.png)
> 
> Tooltip before: ![Screenshot from 2023-08-09 11-53-27](https://user-images.githubusercontent.com/604313/259372976-8633b32d-9b98-4b84-b1aa-0ab1c4557520.png) Tooltip after: ![Screenshot from 2023-08-09 11-53-12](https://user-images.githubusercontent.com/604313/259372981-566cedc2-5c77-4408-be22-e38b15d51e74.png)
> 
> When objects are not loaded yet, spaces are missing: ![Screenshot from 2023-08-09 11-47-59](https://user-images.githubusercontent.com/604313/259372984-a68c924a-4246-462b-9a28-79fd2be32696.png)
> 
> Istio config details page is not loading: ![Screenshot from 2023-08-09 12-06-00](https://user-images.githubusercontent.com/604313/259373075-d5c89c28-12e0-46c1-974e-9acf165c125b.png)

Thanks for the feedback @hhovsepy .

Well the problem with tables is that we need to move all the table component to the new one so the alignment will be fixed after migration. I don't know if make all in the same PR is a good idea .

Something similar behaviour with tooltip

Checking ace-editor. I am going to check how mcuh time is going to take the table migration.

hhovsepy commented:
Config YAML is working fine now, thank you @aljesusg 
![Screenshot from 2023-08-10 12-46-00](https://github.com/kiali/kiali/assets/604313/f81c1f12-6d19-4a88-a202-3d59debd61ac)

hhovsepy commented:
@aljesusg a massive work, I just see some design/width issues:
![Screenshot from 2023-08-21 19-58-49](https://github.com/kiali/kiali/assets/604313/4a3858cf-4600-41f6-85e8-f01a5c1df72f)
![Screenshot from 2023-08-21 19-53-23](https://github.com/kiali/kiali/assets/604313/2b2aaa91-d410-4871-b877-04c140e7cdae)
![Screenshot from 2023-08-21 20-07-15](https://github.com/kiali/kiali/assets/604313/de4c19a5-fff6-4aaa-ba78-507d455edff9)

nrfox reviewed (commented):
The fonts look different when comparing the two side by side. We're they supposed to change so much from v4 --> v5?

![Screenshot from 2023-08-21 14-34-27](https://github.com/kiali/kiali/assets/6226732/12db5c49-6870-40ac-b8df-65fe670f3383)
![Screenshot from 2023-08-21 14-34-35](https://github.com/kiali/kiali/assets/6226732/f100d9b0-0d4d-4b42-b92d-876923177ce0)

nrfox commented on frontend/src/styles/GlobalStyle.ts:
```diff
@@ -4,8 +4,8 @@ export const globalStyle = kialiStyle({
-  // TODO: possible change to --pf-global--FontFamily--redhat-updated--sans-serif
-  fontFamily: 'var(--pf-global--FontFamily--sans-serif)',
+  // TODO: possible change to --pf-v5-global--FontFamily--redhat-updated--sans-serif
+  fontFamily: 'var(--pf-v5-global--FontFamily--sans-serif)',
```
I think this might be the culprit of the weird fonts. Is this var available in pf5? Maybe it needs to be set to something else or it has been split into a few different vars? I can't seem to find pf documentation on this.

aljesusg replied:
yep...they make some changes I am going to try it new options

 --pf-v5-global--FontFamily--text: "RedHatText", helvetica, arial, sans-serif;
  --pf-v5-global--FontFamily--heading: "RedHatDisplay", helvetica, arial, sans-serif;
  --pf-v5-global--FontFamily--monospace: "RedHatMono", "Liberation Mono", consolas, "SFMono-Regular", menlo, monaco, "Courier New", monospace;
  --pf-v5-global--FontFamily--text--vf: "RedHatTextVF", "RedHatText", helvetica, arial, sans-serif;
  --pf-v5-global--FontFamily--heading--vf: "RedHatDisplayVF", "RedHatDisplay", helvetica, arial, sans-serif;
  --pf-v5-global--FontFamily--monospace--vf: "RedHatMonoVF", "RedHatMono", "Liberation Mono", consolas, "SFMono-Regular", menlo, monaco, "Courier New", monospace;
  --pf-v5-global--FontSize--4xl: 2.25rem;
  --pf-v5-global--FontSize--3xl: 1.75rem;
  --pf-v5-global--FontSize--2xl: 1.5rem;
  --pf-v5-global--FontSize--xl: 1.25rem;
  --pf-v5-global--FontSize--lg: 1.125rem;
  --pf-v5-global--FontSize--md: 1rem;
  --pf-v5-global--FontSize--sm: 0.875rem;
  --pf-v5-global--FontSize--xs: 0.75rem;
  --pf-v5-global--FontWeight--normal: 400;
  --pf-v5-global--FontWeight--bold: 700;

aljesusg replied:
The moved from **"RedHatText","Overpass",overpass,helvetica,arial,sans-serif;** to "RedHatText", helvetica, arial, sans-serif. They removed overpass

hhovsepy commented:
Fonts are good now, all my reported issues are fixed.
Only one point, the health icon is smaller than before:
Before:
![Screenshot from 2023-08-22 10-28-04](https://github.com/kiali/kiali/assets/604313/d68c8f09-8fa7-455b-8773-80ac99958d93)
Now:
![Screenshot from 2023-08-22 10-28-15](https://github.com/kiali/kiali/assets/604313/d4cd050a-192f-4c47-9930-8333f91e1f41)

aljesusg commented:
> Fonts are good now, all my reported issues are fixed. Only one point, the health icon is smaller than before: Before: ![Screenshot from 2023-08-22 10-28-04](https://user-images.githubusercontent.com/604313/262285823-d68c8f09-8fa7-455b-8773-80ac99958d93.png) Now: ![Screenshot from 2023-08-22 10-28-15](https://user-images.githubusercontent.com/604313/262285817-d4cd050a-192f-4c47-9930-8333f91e1f41.png)

Ok not worry let me check it. We already know that these migrations hurt hahaha

josunect reviewed (commented):
Great work, @aljesusg ! 

I've just found some small things: 

With pattern fly graph: 

![image](https://github.com/kiali/kiali/assets/49480155/79b9a00b-a789-4d87-96d5-1c2516e3a359)

This is more a suggestion, but in the black mode switch, it doesn't look like a switch, just two separate icons: 

![image](https://github.com/kiali/kiali/assets/49480155/f33161fd-e9ab-44e5-92ce-01c9af899384)

Maybe that was the intention, but as they are related I think it would make sense to somehow create a visual relation in the icons. 

Tested in Firefox and Chrome.

hhovsepy commented:
fyi, still can see the help icon is smaller

aljesusg commented:
> fyi, still can see the help icon is smaller

I didn't change it yet. I am fixing cypress. I'll ping you !!! Thanks!

hhovsepy commented:
> > fyi, still can see the help icon is smaller
> 
> I didn't change it yet. I am fixing cypress. I'll ping you !!! Thanks!

Sorry for the noice.

aljesusg commented:
Fixed icons size @hhovsepy  Thanks !
Fixed fonts and colors sliders
Fixed toolbar darkmode comments @josunect 
![Screenshot from 2023-08-24 12-51-00](https://github.com/kiali/kiali/assets/3019213/c29ba02d-a414-4958-ae5a-1ea6015c20f6)
![Screenshot from 2023-08-24 12-50-30](https://github.com/kiali/kiali/assets/3019213/c9b41f96-2771-4eda-82e9-13911c846b3b)

josunect commented:
> Fixed icons size @hhovsepy Thanks ! Fixed fonts and colors sliders Fixed toolbar darkmode comments @josunect ![Screenshot from 2023-08-24 12-51-00](https://user-images.githubusercontent.com/3019213/262950645-c29ba02d-a414-4958-ae5a-1ea6015c20f6.png) ![Screenshot from 2023-08-24 12-50-30](https://user-images.githubusercontent.com/3019213/262950648-c9b41f96-2771-4eda-82e9-13911c846b3b.png)

Thanks, @aljesusg ! 

One thing, it looks like the patternfly graph still has some styles moved (The find/hide filters, not sure if you also see them incorrectly): 

![image](https://github.com/kiali/kiali/assets/49480155/bedde51f-dec5-48ff-bca9-d85067c23bc2)

hhovsepy reviewed (approved):
LGTM, all issues reported by me are fixed now.

jmazzitelli reviewed (approved):
I built and ran this. I do not see anything broken. All looks good.

josunect reviewed (approved):
LG!
